storage/entities/transaction.js

```diff
@@ -653,6 +664,24 @@ class Transaction extends BaseEntity {
 
 		return filters;
 	}
+
+	/**
+	 * Checks if a dapp exists.
+	 *
+	 * @param {Object} params
+	 * @param {string} params.transactionId
+	 * @param {string} params.name
+	 * @param {string} params.link
+	 * @returns {Promise}
+	 * @todo This method should be replaced with calls to storage get() and correct filters
+	 */
+	getExistingDapps(filters = {}, tx) {
+		const sql = `SELECT name, link FROM dapps
+		WHERE (name = '${filters.name}' OR link = '${
+			filters.link
+		}') AND "transactionId" != '${filters.transactionId_ne}'`;
+		return this.adapter.execute(sql, {}, {}, tx);
+	}
```

Comment:
You don't need this as mentioned in above comment. This logic can be replaced by `isPersisted` with composite filters.

Reply:
OK as this seems that needs to be replaced I will just replace it but it's a similar case for the controller.